caster-editor/src/components/MenuTabPlay.vue

```diff
@@ -0,0 +1,50 @@
+<template>
+  <div>
+    <button
+      class="unstyled"
+      @click="directLink()"
+    >
+      Public
+    </button>
+    <button
+      class="unstyled"
+      @click="debug()"
+    >
+      Debug
+    </button>
+  </div>
+</template>
+
+<script setup lang="ts">
+import type { Graph } from "@/graphql";
+export type GraphEdit = Pick<Graph, "uuid" | "slugName">;
+
+const props = defineProps<{
+  graph: GraphEdit;
+}>();
+
+const directLink = () => {
+  // https://editor.dev.gencaster.org/graph/c5408723-7a2d-4111-826c-9ec507c0b65e
+  // to https://dev.gencaster.org/listen/demo/
+
+  if (import.meta.env.DEV) {
+    const newUrl = "http://127.0.0.1:3000/listen/" + props.graph.slugName;
+    window.open(newUrl, "_blank");
+  } else {
+    const host = window.location.host;
+    const newUrl = "https://" + host + "/listen/" + props.graph.slugName;
```

Comment:
```node
window.location.host
--> "editor.dev.gencaster.org" 
```
This would not work on our deployments - something like

```javascript
window.location.host.split(".").slice(1).join(".")
```

could work though.
But maybe we should be more explicit and use the following rules:

1) **If** the editor is running on `127.0.0.1` as a host, use `127.0.0.1:3000`
2) **else** - if a `ENV` variable `GENCASTER_FRONT_URL` is set, use this
3) **else** - Disable the URL because implicit guessing will lead to errors over time that can be better administrated in a explicit manner.

If we add a list of the URLs of all services to the docker containers via the `vars.env` file we would have one place where all URLs are derived from within our services - https://github.com/Gencaster/gencaster/blob/main/vars.env

What do you think @vin-ni ? If this is OK I can do this b/c the injecting of the env vars to the docker containers is a bit strange currently.

Reply:
Yes, adding all URLS to the global env would be great :)
Let's do that, i will update the frontend code for that.